cluster-install/install.sh

```diff
@@ -2343,8 +2585,28 @@ step_generate_trident_operator_patch() {
     [ -z "$new_image" ] && fatal "no trident operator image found"
 
     logheader $__DEBUG "Generating Trident Operator patch"
-    local -r patch='[{"op":"replace","path":"/spec/template/spec/containers/0/image","value":"'"$new_image"'"}]'
-    _PATCHES_TRIDENT_OPERATOR+=("deploy/trident-operator -n '$namespace' --type=json -p '$(echo "$patch" | jq '.')'")
+    local -r image_patch='{"op":"replace","path":"/spec/template/spec/containers/0/image","value":"'"$new_image"'"}'
+    local -a patch_list="$image_patch"
+
+    # Update image pull secrets if needed
+    if [ -n "$IMAGE_PULL_SECRET" ]; then
+        if echo "$_EXISTING_TRIDENT_OPERATOR_PULL_SECRETS" | grep -q "^${IMAGE_PULL_SECRET}$" &> /dev/null; then
+            logdebug "image pull secret '$IMAGE_PULL_SECRET' already present in trident-operator"
+        else
+            local -r secret_obj='{"name": "'"$IMAGE_PULL_SECRET"'"}'
+            if [ -z "$_EXISTING_TRIDENT_OPERATOR_PULL_SECRETS" ]; then
+                patch_list+=',{"op":"replace","path":"/spec/template/spec/imagePullSecrets","value":['"$secret_obj"']}'
+            else
+                patch_list+=',{"op":"add","path":"/spec/template/spec/imagePullSecrets/-","value":'"$secret_obj"'}'
+            fi
+        fi
+    fi
```

Comment:
This is where we actually generate the trident operator patch that will add the IMAGE_PULL_SECRET, if needed.

Reply:
I think the value is not quite right. When we do the replace, we do want to create an array like you have, but the value in the image pull secrets array looks more like `{"name": "secret-name"}` instead of just "secret-name". I'll make this change here before I merge this PR

Reply:
I merged a fix here, the add and replace operations had a small issue. Should be working now.